trunk/SUAVE/Analyses/Atmospheric/US_Standard_1976.py

```diff
@@ -0,0 +1,172 @@
+        self.update(atmo_data)        
+    
+    def compute_values(self,altitude,type="all"):
+
+        """ Computes values from the International Standard Atmosphere
+
+        Inputs:
+            altitude     : geometric altitude (elevation) (m)
+                           can be a float, list or 1D array of floats
+         
+        Outputs:
+            list of conditions -
+                pressure       : static pressure (Pa)
+                temperature    : static temperature (K)
+                density        : density (kg/m^3)
+                speed_of_sound : speed of sound (m/s)
+                viscosity      : viscosity (kg/m-s)
+            
+        Example:
+            atmosphere = SUAVE.Attributes.Atmospheres.Earth.USStandard1976()
+            atmosphere.ComputeValues(1000).pressure
+          
+        """
+
+        # unpack
+        zs   = altitude
+        gas_base  = self.fluid_properties
+        grav_base = self.planet.sea_level_gravity
+        Rad_base  = self.planet.mean_radius
+        self.fluid_properties = Air()
+        self.planet = Earth()     
+        gas  = self.fluid_properties
+        grav = self.planet.sea_level_gravity        
+        Rad  = self.planet.mean_radius
+        if (gas_base != self.fluid_properties) or (grav_base != self.planet.sea_level_gravity) or (Rad_base != Rad):
+            print 'Warning: US Standard Atmosphere being used outside expected conditions'
+
+        # convert input if necessary
+        if isinstance(zs, int): 
+            zs = np.array([float(zs)])
+        elif isinstance(zs, float):
+            zs = np.array([zs])
+
+        # convert geometric to geopotential altitude
+        zs = zs/(1 + zs/Rad)
+        
+        # Remove redudant brackets
+        if len(zs.shape) == 2:
+            zs = zs[:,0]
+
+        # initialize return data
+        p = np.zeros(np.size(zs))
+        T = np.zeros(np.size(zs))
+        rho = np.zeros(np.size(zs))
+        a = np.zeros(np.size(zs))
+        mew = np.zeros(np.size(zs))
+        z0 = np.zeros(np.size(zs))
+        T0 = np.zeros(np.size(zs))
+        p0 = np.zeros(np.size(zs))
+        alpha = np.zeros(np.size(zs))
+        
+        # evaluate at each altitude
+        zmin = self.breaks.altitude[0]
+        zmax = self.breaks.altitude[-1]
+        
+        if np.amin(zs) < zmin:
+            print "Warning: altitude requested below minimum for this atmospheric model; returning values for h = -2.0 km"
+            zs[zs < zmin] = zmin
+        if np.amax(zs) > zmax:
+            print "Warning: altitude requested above maximum for this atmospheric model; returning values for h = 86.0 km"   
+            zs[zs > zmax] = zmax
+        for i in range(len(self.breaks.altitude)-1): # this uses >= and <= to capture both edges and because values should be the same at the edges
+            z0[(zs>=self.breaks.altitude[i]) & (zs <= self.breaks.altitude[i+1])] = self.breaks.altitude[i]
```

Comment:
super fantastic!  so only have to loop over breaks, not control point.
i guess we get away with this here because the models for different regions of the atmosphere can be described with one equation and different constants.  great find!